File: extensions/button/native/android/ButtonItem.java
package io.neft.extensions.buttonextension;

import android.graphics.Color;
import android.widget.Button;

import io.neft.renderer.NativeItem;
import io.neft.renderer.annotation.OnCreate;
import io.neft.renderer.annotation.OnSet;
import io.neft.utils.ColorValue;

public class ButtonItem extends NativeItem {
    @OnCreate("Button")
    public ButtonItem() {
        super(new Button(APP.getWindowView().getContext()));
        getItemView().setTextColor(Color.BLACK);
    }

    private Button getItemView() {
        return (Button) itemView;
    }

    @OnSet("text")
    public void setText(String val) {
        getItemView().setText(val);
        updateSize();
    }

    @OnSet("textColor")
    public void setTextColor(ColorValue val) {
        getItemView().setTextColor(val.getColor());
    }
}

File: packages/ui/neft-button/native/ios/ButtonItem.swift
import UIKit

extension Extension.Button {
    class ButtonItem: NativeItem {
        override class func main() {
            NativeItemBinding()
                .onCreate("Button") { ButtonItem() }
                .onSet("text") { (item, val: String) in item.setText(val) }
                .onSet("textColor") { (item, val: UIColor?) in item.setTextColor(val) }
                .finalize()
        }

        var buttonView = UIButton()

        init() {
            super.init(itemView: buttonView)
            setTextColor(UIColor.black)
        }

        func setText(_ val: String) {
            buttonView.setTitle(val, for: .normal)
            buttonView.sizeToFit()
            updateSize()
        }

        func setTextColor(_ val: UIColor?) {
            buttonView.setTitleColor(val, for: .normal)
        }
    }
}
